core/src/main/java/tc/oc/pgm/map/MapFactoryImpl.java

```diff
@@ -82,6 +87,14 @@ private void preLoad()
       document.setBaseURI(source.getId());
     }
 
+    // Check for any included map sources, appending them to the document if present
+    Collection<MapInclude> mapIncludes = includes.getMapIncludes(document);
+    if (!mapIncludes.isEmpty()) {
+      for (MapInclude include : mapIncludes) {
+        document.getRootElement().addContent(0, include.getContent());
```

Comment:
Consider including a clone, dont think we have a way to guarantee immutability of xml elements

Reply:
Good idea! Which item should be cloned? The document in map factory or the individual map includes? I believe I made the underlying collection unmodifiable.